src/people.rs

```diff
@@ -894,14 +894,17 @@ pub trait ContextPeopleExt {
     where
         F: Fn(&Context, &[String], usize);
 
-    /// Randomly sample a person from the population.
+    /// Randomly sample a person from the population of people who match the query.
     ///
-    /// This is currently implemented by sampling in `0..current_population`
-    /// but in the future we might have holes where people were removed.
+    /// The syntax here is the same as with [`Context::query_people()`].
     ///
     /// # Errors
     /// Returns `IxaError` if population is 0.
-    fn sample_person<R: RngId + 'static>(&self, rng_id: R) -> Result<PersonId, IxaError>
+    fn sample_person<R: RngId + 'static, T: Query>(
+        &self,
+        rng_id: R,
+        q: T,
```

Comment:
can we name this `query`

Reply:
Sure.

I did think about having two functions, but I think that just having the empty query is less overhead, and it's efficient because I check for it internally.